Packs/SlackITAdmin/Integrations/Slack_IT_Admin/Slack_IT_Admin.py

```diff
@@ -0,0 +1,724 @@
+    )
+
+
+def enable_disable_user_command(client, args):
+    scim = verify_and_load_scim_data(args.get('scim'))
+    parsed_scim = map_scim(scim)
+    user_id = parsed_scim.get('id')
+
+    if not (user_id):
+        return_error('You must provide id of the user')
+
+    if demisto.command() == 'enable-user':
+        format_pre_text = 'Enable'
+        custom_mapping = demisto.params().get('customMappingUpdateUser')
+        slack_scim = client.build_slack_user_profile(args, scim, custom_mapping)
+        slack_scim['active'] = True
+        res = client.update_user(user_id, slack_scim)
+    elif demisto.command() == 'disable-user':
+        format_pre_text = 'Disable'
+        slack_scim = {'active': False}
+        res = client.update_user(user_id, slack_scim)
+
+    res_json = res.json()
+
+    if res.status_code == 200:
+        email = get_primary(res_json.get('emails', []), 'value') if len(res_json.get('emails', [])) > 0 else ""
+        generic_iam_context = OutputContext(success=True, iden=user_id, username=res_json.get('userName'), email=email,
+                                            active=res_json.get('active'), details=res_json)
+    elif res.status_code == 404:
+        generic_iam_context = OutputContext(success=False, iden=user_id, errorCode=404,
+                                            errorMessage=res_json.get('errorSummary'), details=res_json)
+    else:
+        generic_iam_context = OutputContext(success=False, iden=user_id, errorCode=res_json.get('errorCode'),
+                                            errorMessage=res_json.get('errorSummary'), details=res_json)
+
+    generic_iam_context_dt = f'{generic_iam_context.command}(val.id == obj.id && val.instanceName == obj.instanceName)'
+
+    outputs = {
+        generic_iam_context_dt: generic_iam_context.data
+    }
+
+    readable_output = tableToMarkdown(name=f'{format_pre_text} Slack User:',
+                                      t=generic_iam_context.data,
+                                      headers=["brand", "instanceName", "success", "active", "id", "username", "email",
+                                               "errorCode", "errorMessage", "details"],
+                                      removeNull=True)
+    return (
+        readable_output,
+        outputs,
+        generic_iam_context.data
+    )
+
+
+def get_group(client, args):
+    group_id = args.get('groupId')
+    group_name = args.get('groupName')
+
+    if not (group_id or group_name):
+        return_error("You must supply either 'groupId' or 'groupName")
+    if not group_id:
+        group_id = client.get_group_id(group_name)
+
+    res = client.get_group_by_id(group_id)
+    res_json = res.json()
+
+    if res.status_code != 200:
+        return_error(f"Error in API call. Status Code: [{res.status_code}]. Error Response: {res_json}")
+
+    members = res_json.get('members')
+    outputs = {
+        'Slack.Group(val.Id && val.Id === obj.Id)': {
+            'Id': group_id,
```

Comment:
what if group id wasn't supplied?

please check what is supplied - id or name, and accordingly decide which field should be added. If a name was supplied, make sure the key of the output object will be `Slack.Group(val.Name && val.Name === obj.Name)`.

also make sure you are consistent with other commands and with the yml file regarding this issue.

Reply:
code expects either groupId or groupName. If user passes groupName, group_id will be fetched from client.get_group_id method (by API call) and then API will be called to delete group. So variable "group_id" will have value always in this line.